modules/openxr/scene/openxr_composition_layer.cpp

```diff
@@ -103,21 +123,27 @@ void OpenXRCompositionLayer::_create_fallback_node() {
 	should_update_fallback_mesh = true;
 }
 
+void OpenXRCompositionLayer::_remove_fallback_node() {
+	ERR_FAIL_COND(fallback);
+	fallback->queue_free();
+	remove_child(fallback);
+	fallback = nullptr;
+}
+
 void OpenXRCompositionLayer::_on_openxr_session_begun() {
-	if (!is_natively_supported()) {
-		if (!fallback) {
-			_create_fallback_node();
-		}
-	} else if (layer_viewport && is_visible() && is_inside_tree()) {
+	openxr_session_running = true;
+	if (is_natively_supported() && layer_viewport && is_visible() && is_inside_tree()) {
 		openxr_layer_provider->set_viewport(layer_viewport->get_viewport_rid(), layer_viewport->get_size());
 	}
+	if (_should_use_fallback_node() && !fallback) {
+		_create_fallback_node();
+	}
 }
 
 void OpenXRCompositionLayer::_on_openxr_session_stopping() {
-	if (fallback && !Engine::get_singleton()->is_editor_hint()) {
-		fallback->queue_free();
-		remove_child(fallback);
-		fallback = nullptr;
+	openxr_session_running = false;
+	if (fallback && !_should_use_fallback_node()) {
```

Comment:
Possibly safer to just do `if (fallback)`. If there shouldn't be a fallback node, it won't have been created.

This could potentially create a leak where the fallback was created, but since then the dev disabled hole punch, so we don't remove the fallback because now the `_should_use_fallback_node` returns false.

Reply:
Hm. There's really no risk of a leak here, because `fallback` is a node that's a child of the composition layer node, so it'll automatically get cleaned whenever the composition layer node is cleaned up. And, if the dev changed the hole punch setting, it would already have gotten cleaned up in `set_enable_hole_punch()`. But even beyond that, if `_should_use_fallback_node()` now returned `false`, we would still clean it up, because we're checking for `!_should_use_fallback_node()` (negated with `!`) in the above code.

So, I personally think this code is fine as it is.